platform-hub-api/spec/controllers/announcement_templates_controller_spec.rb

```diff
@@ -0,0 +1,270 @@
+require 'rails_helper'
+
+RSpec.describe AnnouncementTemplatesController, type: :controller do
+
+  include_context 'time helpers'
+
+  describe 'GET #index' do
+    it_behaves_like 'unauthenticated not allowed' do
+      before do
+        get :index
+      end
+    end
+
+    it_behaves_like 'authenticated' do
+
+      it_behaves_like 'not an admin so forbidden'  do
+        before do
+          get :index
+        end
+      end
+
+      it_behaves_like 'an admin' do
+
+        before do
+          @announcement_templates = create_list :announcement_template, 3
+        end
+
+        let :total_announcement_templates do
+          @announcement_templates.length
+        end
+
+        let :announcement_template_ids do
+          @announcement_templates.sort_by(&:shortname).map(&:friendly_id)
+        end
+
+        it 'should return a list of all announcement templates' do
+          get :index
+          expect(response).to be_success
+          expect(json_response.length).to eq total_announcement_templates
+          expect(pluck_from_json_response('id')).to eq announcement_template_ids
```

Comment:
Do you always expect result in particular order? If not then maybe change matcher type.